Lecture2/Task2/TaskTwoMain.java

```diff
@@ -0,0 +1,62 @@
+import java.util.*;
+
+public class TaskTwoMain {
+	final static String FIRST_LETTER = "M";
+	private static double average;
+
+	public static void main(String[] args) {
+		List<Student> students = new ArrayList<>();
+
+		students.add(new Student(1, 60, "Rudolf", "Schenker"));
+		students.add(new Student(2, 75, "Klaus", "Meine"));
+		students.add(new Student(3, 55, "Matthias", "Jabs"));
+		students.add(new Student(4, 58, "Paweł", "Mąciwoda"));
+		students.add(new Student(5, 64, "Mikkey", "Dee"));
+
+		Collections.sort(students);
+		System.out.println("List sorted by Age:");
+		for (Student student : students) {
+			System.out.println(student.getAge() + " " + student.getName());
+		}
+
+		System.out.println(String.format("First letter of the name is %s:", FIRST_LETTER));
+		Filter.getStudentsM(students, FIRST_LETTER);
+
+		System.out.println(String.format("Average age of students is:\n%s",
+				getAvarageAge(average, students)));
+
+		System.out.println("Names from the Map:");
+		Map<Integer, Student> studentsMap = getStudentsListToMap(students);
+		for (Student key : studentsMap.values()) {
+			System.out.println(key.getName());
+		}
+
+		System.out.println("Names from the Map (id > 3):");
+		Map<Integer, Student> studentMapAboveThree = new HashMap<>();
+		for (Student key : studentsMap.values()) {
+			if (key.getId() > 3) {
+				studentMapAboveThree.put(key.getId(), key);
+				System.out.println(key.getName());
+			}
+		}
+	}
+
+	private static double getAvarageAge(double average, List<Student> students) {
```

Comment:
please create one common interface with methods `getAvarageAge ` , `getStudentsListToMap `  and other
create two implementation for stream/non stream 

and use this interface in task 3 and 4

Reply:
done for 2